pkg/operator/ceph/file/mds/spec.go

```diff
@@ -65,6 +76,11 @@ func (c *Cluster) makeDeployment(mdsConfig *mdsConfig) *apps.Deployment {
 	c.fs.Spec.MetadataServer.Annotations.ApplyToObjectMeta(&podSpec.ObjectMeta)
 	c.fs.Spec.MetadataServer.Placement.ApplyToPodSpec(&podSpec.Spec)
 
+	if mdsConfig.FirstInZone {
```

Comment:
Can't we always append the zone anti-affinity? As long as we use the preferredDuringScheduling, K8s should just ignore it if there aren't enough zones  or if we have multiple MDSs in each zone. This will simplify things so we shouldn't need to detect how many nodes are in a zone and try and detect the scheduling decisions ourselves. We should let k8s do all these scheduling detection and decisions. We're just adding the anti-affinity as a preferred setting so k8s can do the hard work for us.